### PR TITLE
Add missing SQA doco for python packages

### PR DESCRIPTION
**Pull request by GiudGiud**
1300 warnings -> 1100 left
refs #8661 #7451 #15968

### CODE REVIEW

**Comment by cticenhour** [comment]
@GiudGiud added to my list for review tomorrow.

File: python/chigger/tests/viewports/tests [hunk 1]
```diff
@@ -1,8 +1,11 @@
 [Tests]
+  issues = '#8661'
+  design = 'chigger/index.md'
   [viewports]
     type = ImageDiff
     command = viewports.py
     imagediff = 'viewports.png'
     display_required = true
+    requirement = 'The system shall be able to specify the coordinates of a view port in the results.'
```
Comment (cticenhour) [inline python/chigger/tests/viewports/tests:9]:
```suggestion
    requirement = 'The system shall be able allow the user to specify the coordinates of a view port in the results.'
```

Should it say this instead? Not familiar with this feature.

File: python/chigger/tests/writers/tests [hunk 2]
```diff
@@ -1,8 +1,11 @@
 [Tests]
+  issues = '#8661'
+  design = 'chigger/index.md'
   [writers]
     type = PythonUnitTest
     input = writers.py
     buffer = True
     display_required = true
+    requirement = 'The system shall be able to write plot with various formats, and error if an unsupported format is requested.'
```
Comment (cticenhour) [inline python/chigger/tests/writers/tests:9]:
```suggestion
    requirement = 'The system shall be able to write a plot with various formats, and error if an unsupported format is requested.'
```

File: python/parameters/test/tests [hunk 3]
```diff
@@ -1,10 +1,14 @@
 [Tests]
+  issues = '#12095'
+  design = 'python/index.md'
   [Parameter]
     type = PythonUnitTest
     input = test_Parameter.py
+    requirement = "The system shall be able to define parameters in Python, similar to individual MooseObject parameters in C++, with numerous similar capabilities, including a name, a type, allowed and default values, descriptions, required and private status."
```
Comment (cticenhour) [inline python/parameters/test/tests:7]:
```suggestion
    requirement = "The system shall be able to define parameters and associated metadata in Python that allows for type-checking, restrictions, defaults, verification, and documentation."
```

File: python/parameters/test/tests [hunk 4]
```diff
@@ -1,10 +1,14 @@
...
   []
   [InputParameters]
     type = PythonUnitTest
     input = test_InputParameters.py
+    requirement = "The system shall be able to define groups of parameters to Python objects, similar to the parameters of a MooseObject in C++, with numerous similar capabilities, including parameter errors, the ability to retrieve parameters, to set defaults, and others."
```
Comment (cticenhour) [inline python/parameters/test/tests:12]:
```suggestion
    requirement = "The system shall have a warehouse and capability to manage groups of parameters in Python objects, including parameter errors and the ability to retrieve and set parameters, to set defaults, and to validate parameters."
```

File: python/peacock/tests/base/tests [hunk 5]
```diff
@@ -1,12 +1,16 @@
...
     type = RunApp
     command = test_PluginManager.py
     display_required = true
+    requirement = 'The system shall have a graphical user interface which supports Qt plugins.'
   []
   [collapsible]
     type = RunApp
     command = test_PeacockCollapsibleWidget.py
     display_required = true
+    requirement = 'The system shall have a graphical user interface which supports collapside widgets.'
```
Comment (cticenhour) [inline python/peacock/tests/base/tests:14]:
```suggestion
    requirement = 'The system shall have a graphical user interface which supports collapsible widgets.'
```

File: python/peacock/tests/execute_tab/ConsoleOutputViewerPlugin/tests [hunk 6]
```diff
@@ -1,7 +1,10 @@
 [Tests]
+  issues = '#7451'
+  design = 'python/peacock.md'
   [ConsoleOutputViewerPlugin]
     type = PythonUnitTest
     input = test_ConsoleOutputViewerPlugin.py
     display_required = true
+    requirement = "The system shall include a graphical interface that display scrollable and auto-scrollable console output."
```
Comment (cticenhour) [inline python/peacock/tests/execute_tab/ConsoleOutputViewerPlugin/tests:8]:
```suggestion
    requirement = "The system shall include a graphical interface that displays scrollable and auto-scrollable console output."
```

File: python/peacock/tests/execute_tab/ExecuteOptionsPlugin/tests [hunk 7]
```diff
@@ -1,7 +1,10 @@
 [Tests]
+  issues = '#7451'
+  design = 'python/peacock.md'
   [ExecuteOptionsPlugin]
     type = PythonUnitTest
     input = test_ExecuteOptionsPlugin.py
     display_required = true
+    requirement = "The system shall include a graphical interface that lets user specifies execution options such as the number of threads, parallel processes, working directory and input file name."
```
Comment (cticenhour) [inline python/peacock/tests/execute_tab/ExecuteOptionsPlugin/tests:8]:
```suggestion
    requirement = "The system shall include a graphical interface that lets users specify execution options such as the number of threads, parallel processes, working directory and input file name."
```

File: python/peacock/tests/execute_tab/ExecuteTabPlugin/tests [hunk 8]
```diff
@@ -1,7 +1,10 @@
 [Tests]
+  issues = '#7451'
+  design = 'python/peacock.md'
   [ExecuteTabPlugin]
     type = PythonUnitTest
     input = test_ExecuteTabPlugin.py
     display_required = true
+    requirement = "The system shall include a graphical interface that has a plugin execution tab, that lets user specify execution parameters, through the command line, programmatically and on the interface, and set these run option preferences for individual plugins."
```
Comment (cticenhour) [inline python/peacock/tests/execute_tab/ExecuteTabPlugin/tests:8]:
```suggestion
    requirement = "The system shall include a graphical interface that has a plugin execution tab that lets user specify execution parameters, through the command line, programmatically and on the interface, and set these run option preferences for individual plugins."
```

File: python/peacock/tests/exodus_tab/tests [hunk 9]
```diff
@@ -1,93 +1,110 @@
 [Tests]
+  issues = '#7451'
+  design = 'python/peacock.md'
   [VTKWindowPlugin]
     type = PythonUnitTest
     input = test_VTKWindowPlugin.py
     separate = True
     display_required = true
+    requirement = "The system shall include a graphical interface that can display both mesh and solution Exodus files and let the user, including series of files created with mesh adaptivity."
```
Comment (cticenhour) [inline python/peacock/tests/exodus_tab/tests:9]: Let the user what?

File: python/peacock/tests/input_tab/BlockEditor/tests [hunk 10]
```diff
@@ -1,7 +1,10 @@
 [Tests]
+  issues = '#7451'
+  design = 'python/peacock.md'
   [BlockEditor]
     type = PythonUnitTest
     input = test_BlockEditor.py
     display_required = true
+    requirement = "The system shall include a graphical interface that lets the user modify input files for example by adding or cloning blocks, specifying their parameters, handling the types of the parameters, and handling regex 'star' syntax."
```
Comment (cticenhour) [inline python/peacock/tests/input_tab/BlockEditor/tests:8]:
```suggestion
    requirement = "The system shall include a graphical interface that lets the user modify input files, for example, by adding or cloning blocks, specifying their parameters, handling the types of the parameters, and handling regex 'star' syntax."
```

File: python/peacock/tests/input_tab/BlockInfo/tests [hunk 11]
```diff
@@ -1,7 +1,10 @@
 [Tests]
+  issues = '#7451'
+  design = 'python/peacock.md'
   [BlockInfo]
     type = PythonUnitTest
     input = test_BlockInfo.py
     display_required = true
+    requirement = "The system shall include a graphical interface that can gather and process parameters for input files blocks, notably their type, name, input path and value."
```
Comment (cticenhour) [inline python/peacock/tests/input_tab/BlockInfo/tests:8]:
```suggestion
    requirement = "The system shall include a graphical interface that can gather and process parameters for input file blocks - notably their type, name, input path and value."
```

File: python/peacock/tests/input_tab/BlockTree/tests [hunk 12]
```diff
@@ -1,7 +1,10 @@
 [Tests]
+  issues = '#7451'
+  design = 'python/peacock.md'
   [BlockTree]
     type = PythonUnitTest
     input = test_BlockTree.py
     display_required = true
+    requirement = "The system shall include a graphical interface that can store an input file as a tree of blocks and lets user modify individual blocks."
```
Comment (cticenhour) [inline python/peacock/tests/input_tab/BlockTree/tests:8]:
```suggestion
    requirement = "The system shall include a graphical interface that can store an input file as a tree of blocks and lets users modify individual blocks."
```

File: python/peacock/tests/input_tab/CommentEditor/tests [hunk 13]
```diff
@@ -1,7 +1,10 @@
 [Tests]
+  issues = '#7451'
+  design = 'python/peacock.md'
   [CommentEditor]
     type = PythonUnitTest
     input = test_CommentEditor.py
     display_required = true
+    requirement = "The system shall include a graphical interface that lets user specify comments in their input files."
```
Comment (cticenhour) [inline python/peacock/tests/input_tab/CommentEditor/tests:8]:
```suggestion
    requirement = "The system shall include a graphical interface that lets users specify comments in their input files."
```

File: python/peacock/tests/input_tab/InputFile/tests [hunk 14]
```diff
@@ -1,7 +1,10 @@
 [Tests]
+  issues = '#7451'
+  design = 'python/peacock.md'
   [InputFile]
     type = PythonUnitTest
     input = test_InputFile.py
     display_required = true
+    requirement = "The system shall include a graphical interface that can read user inputs and error if they are missing, present invalid duplicate syntax, or an unsupported file extension."
```
Comment (cticenhour) [inline python/peacock/tests/input_tab/InputFile/tests:8]:
```suggestion
    requirement = "The system shall include a graphical interface that can read user inputs and error if they are missing, present invalid duplicate syntax, or are an unsupported file extension."
```

File: python/peacock/tests/input_tab/InputFileEditor/tests [hunk 15]
```diff
@@ -1,7 +1,10 @@
 [Tests]
+  issues = '#7451'
+  design = 'python/peacock.md'
   [InputFileEditor]
     type = PythonUnitTest
     input = test_InputFileEditor.py
     display_required = true
+    requirement = "The system shall include a graphical interface that lets user create new inputs files, including new blocks and suggesting missing required blocks."
```
Comment (cticenhour) [inline python/peacock/tests/input_tab/InputFileEditor/tests:8]:
```suggestion
    requirement = "The system shall include a graphical interface that lets users create new input files, including adding new blocks and suggesting missing required blocks."
```

File: python/peacock/tests/input_tab/InputFileEditorPlugin/tests [hunk 16]
```diff
@@ -1,7 +1,10 @@
 [Tests]
+  issues = '#7451'
+  design = 'python/peacock.md'
   [InputFileEditorPlugin]
     type = PythonUnitTest
     input = test_InputFileEditorPlugin.py
     display_required = true
+    requirement = "The system shall include a graphical interface that lets user modify input files inside widgets."
```
Comment (cticenhour) [inline python/peacock/tests/input_tab/InputFileEditorPlugin/tests:8]:
```suggestion
    requirement = "The system shall include a graphical interface that lets users modify input files inside widgets."
```

File: python/peacock/tests/input_tab/InputFileEditorWithMesh/tests [hunk 17]
```diff
@@ -1,8 +1,11 @@
 [Tests]
+  issues = '#7451'
+  design = 'python/peacock.md'
   [InputFileEditorWithMesh]
     type = PythonUnitTest
     input = test_InputFileEditorWithMesh.py
     separate = True
     display_required = true
+    requirement = "The system shall include a graphical interface that lets user modify input files with an existing mesh, pulling in block information from the mesh and highlighting blocks and boundaries in the mesh when specifying them as parameters."
```
Comment (cticenhour) [inline python/peacock/tests/input_tab/InputFileEditorWithMesh/tests:9]:
```suggestion
    requirement = "The system shall include a graphical interface that lets users modify input files with an existing mesh, pulling in block information from the mesh and highlighting blocks and boundaries in the mesh when specifying them as parameters."
```

File: python/peacock/tests/input_tab/OutputNames/tests [hunk 18]
```diff
@@ -1,7 +1,10 @@
 [Tests]
+  issues = '#7451'
+  design = 'python/peacock.md'
   [Outputnames]
     type = PythonUnitTest
     input = test_OutputNames.py
     display_required = true
+    requirement = "The system shall include a graphical interface that lets user specify new file names for the application output, and follows expected name conventions otherwise."
```
Comment (cticenhour) [inline python/peacock/tests/input_tab/OutputNames/tests:8]:
```suggestion
    requirement = "The system shall include a graphical interface that lets users specify new file names for the application output and follows default name conventions when a new name is not specified."
```

File: python/peacock/tests/input_tab/ParameterInfo/tests [hunk 19]
```diff
@@ -1,7 +1,10 @@
 [Tests]
+  issues = '#7451'
+  design = 'python/peacock.md'
   [ParameterInfo]
     type = PythonUnitTest
     input = test_ParameterInfo.py
     display_required = true
+    requirement = "The system shall include a graphical interface that can read and store parameter information, such as for example the parameter name, description, values, C++ and Python value types."
```
Comment (cticenhour) [inline python/peacock/tests/input_tab/ParameterInfo/tests:8]:
```suggestion
    requirement = "The system shall include a graphical interface that can read and store parameter information, such as, for example, the parameter name, description, values, C++ and Python value types."
```

File: python/peacock/tests/peacock_app/check_exodus_state/tests [hunk 20]
```diff
@@ -1,8 +1,11 @@
 [Tests]
+  issues = '#7451'
+  design = 'python/peacock.md'
   [test]
     type = PythonUnitTest
     input = test_exodus_state.py
     separate = True
     display_required = true
+    requirement = "The system shall include a graphical interface which maintains its exodus file viewing settings when a simulation is executed several times."
```
Comment (cticenhour) [inline python/peacock/tests/peacock_app/check_exodus_state/tests:9]:
```suggestion
    requirement = "The system shall include a graphical interface which maintains its Exodus file viewing settings when a simulation is executed several times."
```

File: python/peacock/tests/utils/tests [hunk 21]
```diff
@@ -1,37 +1,46 @@
...
     type = PythonUnitTest
     input = test_TerminalUtils.py
     display_required = true
+    requirement = "The system shall include a graphical interface that can output terminal output to HTML, notably preserving text color."
   []
   [Widgetutils]
     type = PythonUnitTest
     input = test_WidgetUtils.py
     display_required = true
+    requirement = "The system shall include a graphical interface that can create widgets with notably buttons, labels and a progress bar."
```
Comment (cticenhour) [inline python/peacock/tests/utils/tests:38]:
```suggestion
    requirement = "The system shall include a graphical interface that can create widgets with buttons, labels and a progress bar."
```

File: python/chigger/tests/speed/speed.py [hunk 22]
```diff
@@ -12,8 +12,13 @@
...
 import time
 import chigger
+import mooseutils
 
-filename = os.path.join(os.getenv('MOOSE_DIR'), 'modules', 'phase_field', 'examples', 'grain_growth', 'grain_growth_2D_graintracker_out.e')
+if os.getenv('MOOSE_DIR') == None:
+    mooseutils.mooseError("MOOSE_DIR must be set in this script")
+
+# filename = os.path.join(os.getenv('MOOSE_DIR'), 'modules', 'phase_field', 'examples', 'grain_growth', 'grain_growth_2D_graintracker_out.e')
+filename = os.path.join(os.getenv('MOOSE_DIR'), 'modules', 'phase_field', 'test', 'tests', 'grain_growth', 'gold', 'test_out.e-s005')
```
Comment (cticenhour) [inline python/chigger/tests/speed/speed.py:21]: Why this change, and did you forget to remove the commented out previous line?

**Review by cticenhour — CHANGES_REQUESTED** [review]

File: python/chigger/tests/global_data/tests [hunk 23]
```diff
@@ -0,0 +1,36 @@
...
+    display_required = true
+    requirement = "The system shall be able to load global data at various time steps."
+  []
+
+  [plot_global_data]
+    type = ImageDiff
+    command = plot_global_data.py
+    imagediff = 'plot_global_data.png'
+    display_required = true
+    requirement = "The system shall be able to draw line plots from global data."
```
Comment (GiudGiud) [inline python/chigger/tests/global_data/tests:25]: I ll double check here. I thought I changed because the test actually does field data.

File: python/chigger/tests/speed/speed.py (hunk 22 above)
Comment (GiudGiud) [reply python/chigger/tests/speed/speed.py:21]: the original input is not included in the repository. it only appears after running the test suite for phase field. I ll remove the previous line

File: python/chigger/tests/global_data/tests (hunk 23 above)
Comment (GiudGiud) [reply python/chigger/tests/global_data/tests:25]: seems it was global data. not sure what happened.

File: python/chigger/tests/speed/speed.py (hunk 22 above)
Comment (GiudGiud) [reply python/chigger/tests/speed/speed.py:21]: Removed the previous line and a few commented lines in the input

File: python/peacock/tests/exodus_tab/tests (hunk 9 above)
Comment (GiudGiud) [reply python/peacock/tests/exodus_tab/tests:9]: Added a list.
These tests are very not-granular. I m sure it saved some time building the test suite but it makes for off requirements

File: python/chigger/tests/rotational_extrusion/tests [hunk 24]
```diff
@@ -1,4 +1,6 @@
 [Tests]
+  issues = '#8661'
+  design = 'chigger/index.md'
   [exodus]
     type = ImageDiff
     command = rotational_extrusion.py
```
[the remark this answers is not in the document]
Comment (GiudGiud) [reply python/chigger/tests/rotational_extrusion/tests:6]: done, and for the other tests as well

**Comment by GiudGiud** [comment]
failure unrelated (torch)

File: python/chigger/tests/line/tests [hunk 25]
```diff
@@ -27,7 +26,7 @@
     command = empty.py
     imagediff = 'empty.png'
     allowed_linux = 0.935
-    requirement = 'An empty graph shall be a valid chigger widget'
+    requirement = 'An empty graph shall be a valid The system widget'
```
Comment (cticenhour) [inline python/chigger/tests/line/tests:29]:
```suggestion
    requirement = 'The system shall allow an empty graph to be a valid widget'
```

Bad find and replace.

File: python/peacock/tests/execute_tab/ExecuteTabPlugin/tests [hunk 26]
```diff
@@ -1,7 +1,10 @@
 [Tests]
+  issues = '#7451'
+  design = 'python/peacock.md'
   [ExecuteTabPlugin]
     type = PythonUnitTest
     input = test_ExecuteTabPlugin.py
     display_required = true
+    requirement = "The system shall include a graphical interface that has a plugin execution tab that lets user specify execution parameters, through the command line, programmatically and on the interface, and set these run option preferences for individual plugins."
```
Comment (cticenhour) [inline python/peacock/tests/execute_tab/ExecuteTabPlugin/tests:8]:
```suggestion
    requirement = "The system shall include a graphical interface that has a plugin execution tab that lets users specify execution parameters, through the command line, programmatically and on the interface, and set these run option preferences for individual plugins."
```

**Review by cticenhour — CHANGES_REQUESTED** [review]
A couple more things, but I think after this it is ready to go.

**Review by cticenhour — APPROVED** [review]

**Comment by GiudGiud** [comment]
Failures unrelated, due to current work moving an item from BISON to Moose

**Comment by GiudGiud** [comment]
Thanks for the review!
